server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+
+        mock_request = mock.MagicMock()
+        mock_data = {'delta': {'description': 'test'}}
+        mock_request.body = json.dumps(mock_data)
+
+        repos_resource = RepoResourceView()
+        response = repos_resource.put(mock_request, 'mock_repo')
+
+        mock_repo.update(_href='/mock/path/')
+        mock_manager.update_repo_and_plugins.assert_called_once_with(
+            'mock_repo', mock_data['delta'], None, None)
+        mock_resp.assert_called_once_with(mock_repo)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories._convert_repo_dates_to_strings')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_put_existing_repo_with_importer(self, mock_factory, mock_rev, mock_date, mock_resp):
+        """
+        Test update with importer config update.
+        """
+
+        mock_manager = mock_factory.repo_manager.return_value
+        mock_task_result = mock_manager.update_repo_and_plugins.return_value
+        mock_task_result.spawned_tasks = False
+        mock_task_result.serialize.return_value = {'mock': 'repo'}
+
+        mock_request = mock.MagicMock()
+        mock_data = {'importer_config': 'importer_data'}
+        mock_request.body = json.dumps(mock_data)
+
+        repos_resource = RepoResourceView()
+        response = repos_resource.put(mock_request, 'mock_repo')
+
+        mock_resp.assert_called_once_with({'mock': 'repo'})
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories._convert_repo_dates_to_strings')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_put_existing_repo_with_distributor(self, mock_factory, mock_rev, mock_date, mock_resp):
+        """
+        Test update with importer config update.
+        """
+
+        mock_manager = mock_factory.repo_manager.return_value
+        mock_task_result = mock_manager.update_repo_and_plugins.return_value
+        mock_task_result.spawned_tasks = 'distributor'
+
+        mock_request = mock.MagicMock()
+        mock_data = {'distributor_configs': 'distributor_data'}
+        mock_request.body = json.dumps(mock_data)
+
+        repos_resource = RepoResourceView()
+        try:
```

Comment:
i think you can avoid try/except block just by doing 
self.assertRaises(OperationPostponed, repos_resource.put, request, 'mock_repo')